rocks-db/src/bin/fork_detector/main.rs

```diff
@@ -0,0 +1,234 @@
+use entities::models::AssetSignature;
+use metrics_utils::red::RequestErrorDurationMetrics;
+use rocks_db::cl_items::{ClItemKey, ClLeafKey};
+use rocks_db::column::TypedColumn;
+use rocks_db::migrator::MigrationState;
+use rocks_db::Storage;
+use solana_sdk::pubkey::Pubkey;
+use std::env;
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::Instant;
+use tokio::sync::Mutex;
+use tokio::task::JoinSet;
+
+const BATCH_TO_DROP: usize = 1000;
+
+#[tokio::main(flavor = "multi_thread")]
+pub async fn main() -> Result<(), String> {
+    // Retrieve the database paths from command-line arguments
+    let args: Vec<String> = env::args().collect();
+    if args.len() < 2 {
+        println!("Usage: {} <source_db_path>", args[0]);
+        std::process::exit(1);
+    }
+    let source_db_path = &args[1];
+
+    println!("Start looking for forks...");
+
+    // Copy specified column families
+    if let Err(e) = find_forks(source_db_path).await {
+        println!("Failed to find and delete forks: {}.", e);
+    } else {
+        println!("Forks dropped successfully.");
+    }
+
+    Ok(())
+}
+
+async fn find_forks(source_path: &str) -> Result<(), String> {
+    let red_metrics = Arc::new(RequestErrorDurationMetrics::new());
+    // Open source and destination databases
+    let start = Instant::now();
+
+    println!("Opening DB...");
+
+    let source_db = Storage::open(
+        source_path,
+        Arc::new(Mutex::new(JoinSet::new())),
+        red_metrics.clone(),
+        MigrationState::Last,
+    )
+    .map_err(|e| e.to_string())?;
+
+    println!("Opened for {:?}", start.elapsed());
+
+    println!("Iterating over column family...");
+
+    let start = Instant::now();
+    let all_assets_signatures = source_db.asset_signature.iter_start();
+
+    // vec[(signature, sequence, slot)]
+    let mut signatures: Vec<(String, u64, u64)> = Vec::new();
+    // (treeId, leaf_idx)
+    let mut current_asset = ("".to_string(), 0);
+
+    // vec[(treeId, sequence)]
+    let mut sequences_to_delete = Vec::new();
+
+    for sig in all_assets_signatures {
+        match sig {
+            Ok((k, v)) => {
+                let key = AssetSignature::decode_key(k.to_vec()).unwrap();
+                let value = bincode::deserialize::<AssetSignature>(v.as_ref()).unwrap();
+
+                let asset_identifier = (key.tree.to_string(), key.leaf_idx);
+
+                // once got new asset check signatures for previous one
+                if asset_identifier != current_asset {
+                    // more than 2 actions with asset
+                    if signatures.len() >= 2 {
+                        // check only last two because if there was some forked tx
```

Comment:
transaction with 13th sequence brings latest valid state, so everything is ok here.
but if that last transaction didn't happen it means that we have data in DB with seq 12 from forked block, which is a problem because by fact that updates is marked with seq 10 in the main chain.